### DISCUSSION TITLE
Github Workflow 설정 질문

### DISCUSSION THREAD

**Discussion by mirusu400** [klas-helper/klas-helper-extension#15] [Ideas]
@nbsp1221 

현재 Github Workflow가 린팅 및 빌드를 진행하는 Workflow인듯한데, 그 결과를 저장소에 커밋 및 푸시하고 있지 않아서 결론적으로 아무런 동작을 하지 않고 있습니다.

의도하신게 린팅/빌드 후 자동으로 저장소에 저장하시는게 맞다면 커밋/푸시하는 로직도 Workflow에 포함되어야 할듯합니다

**Comment by nbsp1221** [comment]
의도했던 것은 린트와 빌드가 정상적으로 동작하는지 확인하기 위해 workflow를 셋업했던 거라 커밋/푸시하는 로직은 일부로 포함시키진 않았습니다.

현재는 KLAS Helper 메인 로직만 따로 빌드하여 결과물을 커밋해 관리하고 있는데, 나중에는 확장 프로그램의 로직도 포함한 모든 코드를 함께 빌드한 후 결과물로 만들어 익스텐션으로 바로 배포하고 커밋으로는 별도로 관리하지 않는 방향을 생각해보고 있습니다.

하지만 미래의 이야기이긴 해서 지금 상황에 맞도록 workflow에 커밋/푸시하는 로직을 일단 추가해놔도 괜찮을 것 같긴 한데 어떻게 하는 게 좋을까요?

**Reply by mirusu400** [reply]
자동으로 빌드해 결과물 `main-ext.js`파일을 커밋/푸시하는건 저도 좋은 생각인것 같습니다.

이와는 별개로, 바로 익스텐션 배포하는 방안도 생각 해봤는데 크롬 확장프로그램 스토어에 올리는과정이 조금 복잡해서 난관이 있어보입니다.

지금 당장은
1. 필요없는 파일들 삭제 (src 및 update 폴더, git 히든 폴더 등)
2. zip파일로 압축
3. 크롬 확장 앱스토어에 업로드
4. Changelog 갱신 및 등록
5. 검토 과정 끝난 뒤 자동 업데이트 (해당 과정은 크롬 앱스토어 측에서 자동으로 해줌)

이정도 작업을 하기위해선 기존에 크롬 확장 프로그램 CICD작업을 위해 공개된 쉘파일을 약간 수정해서 사용해야 될듯합니다

**Reply by nbsp1221** [reply]
좋습니다. 그러면 workflow는 자동으로 커밋/푸시까지 되도록 수정하는 것으로 하겠습니다.

익스텐션 배포의 경우 처음부터 CI/CD 작업을 하는 것은 아무래도 힘들 것 같고 단계적으로

1. 빌드에 익스텐션 파일까지 포함. 나온 결과물을 사람이 직접 배포하고, 커밋에서는 제외.
2. CI/CD를 셋업해서 사람이 직접 배포하는 과정을 없앰.

처럼 개선해 나가면 될 것 같습니다.
